laplace/baselaplace.py

```diff
@@ -1792,8 +1881,1041 @@ def sample(
+
+    This class implements the (approximate) GP inference through which
+    we obtain the desired quantities (posterior predictive, marginal log-likelihood).
+    See [Improving predictions of Bayesian neural nets via local linearization (Immer et al., 2021)](https://arxiv.org/abs/2008.08400)
+    for more details.
+
+    Note that for `likelihood='classification'`, we approximate \( L_{NN} \\) with a diagonal matrix
+    ( \\( L_{NN} \\) is a block-diagonal matrix, where blocks represent Hessians of per-data-point log-likelihood w.r.t.
+    neural network output \\( f \\), See Appendix [A.2.1](https://arxiv.org/abs/2008.08400) for exact definition). We
+    resort to such an approximation because of the (possible) errors found in Laplace approximation for
+    multiclass GP classification in Chapter 3.5 of [R&W 2006 GP book](http://www.gaussianprocess.org/gpml/),
+    see the question
+    [here](https://stats.stackexchange.com/questions/555183/gaussian-processes-multi-class-laplace-approximation)
+    for more details. Alternatively, one could also resort to *one-vs-one* or *one-vs-rest* implementations
+    for multiclass classification, however, that is not (yet) supported here.
+
+    Parameters
+    ----------
+    num_data : int
+        number of data points for Subset-of-Data (SOD) approximate GP inference.
+    diagonal_kernel : bool
+        GP kernel here is product of Jacobians, which results in a \\( C \\times C\\) matrix where \\(C\\) is the output
+        dimension. If `diagonal_kernel=True`, only a diagonal of a GP kernel is used. This is (somewhat) equivalent to
+        assuming independent GPs across output channels.
+
+    See `BaseLaplace` class for the full interface.
+    """
+
+    # key to map to correct subclass of BaseLaplace, (subset of weights, Hessian structure)
+    _key = ("all", "gp")
+
+    def __init__(
+        self,
+        model: nn.Module,
+        likelihood: Likelihood | str,
+        num_data: int,
+        sigma_noise: float | torch.Tensor = 1.0,
+        prior_precision: float | torch.Tensor = 1.0,
+        prior_mean: float | torch.Tensor = 0.0,
+        temperature: float = 1.0,
+        enable_backprop: bool = False,
+        dict_key_x="inputs_id",
+        dict_key_y="labels",
+        backend: type[CurvatureInterface] | None = BackPackGGN,
+        backend_kwargs: dict[str, Any] | None = None,
+        diagonal_kernel: bool = False,
+        seed: int = 0,
+    ):
+        assert backend in [BackPackGGN, AsdlGGN, CurvlinopsGGN]
+        self._check_prior_precision(prior_precision)
+        super().__init__(
+            model,
+            likelihood,
+            sigma_noise,
+            prior_precision,
+            prior_mean,
+            temperature,
+            enable_backprop,
+            dict_key_x,
+            dict_key_y,
+            backend,
+            backend_kwargs,
+        )
+        self.enable_backprop = enable_backprop
+
+        self.num_data = num_data
```

Comment:
Would call this `n_data_sod`, `n_data_subset`, or `n_subset`. Later, you compare `n_data` to `num_data` and that is confusing since both refer potentially to the same thing.

Reply:
Totally agree, I think `n_subset` is a good name for the variable.